app/src/main/java/com/lla/panicbutton/ui/fragments/onboardingfragments/ViewPagerFragment.kt

```diff
@@ -0,0 +1,79 @@
+package com.lla.panicbutton.ui.fragments.onboardingfragments
+
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.widget.ViewPager2
+import com.lla.panicbutton.R
+import com.lla.panicbutton.adapters.ViewPagerAdapter
+import com.tbuonomo.viewpagerdotsindicator.SpringDotsIndicator
+import kotlinx.android.synthetic.main.fragment_view_pager.view.*
+
+class ViewPagerFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_view_pager, container, false)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        val springDotsIndicator = requireView().findViewById<SpringDotsIndicator>(R.id.spring_dots_indicator)
+        val viewPager = requireView().findViewById<ViewPager2>(R.id.viewPager2)
+        springDotsIndicator?.setViewPager2(viewPager)
+
+        view.viewPagerNextButton.setOnClickListener{
+            if (viewPager.currentItem+1 < adapter.itemCount){
+                viewPager.currentItem+=1
+            }else{
```

Comment:
some weird spacing, is the formatter not catching these? I think you can install "save actions" on android studio which is a plugin with features, then you can have "format changed lines" in it so it can format, lemme know if you need help setting that up, maybe there's a YouTube video for it.

Reply:
![image](https://user-images.githubusercontent.com/71042901/125206002-da253800-e252-11eb-87a0-ebececfb9e47.png)

In Save Actions settings, do I check the other two as well?

Reply:
no this is good! I usually do "reformat only changed code" just because sometimes it adds a lot of changes if you format old code but it's good for a small project.